tests/Integration/Subscriber/Cache/PurgeActionsSubscriber/purgeUserCache.php

```diff
@@ -0,0 +1,58 @@
+<?php
+namespace WP_Rocket\Tests\Integration\Subscriber\Cache\PurgeActionsSubscriber;
+
+use WP_Rocket\Tests\Integration\TestCase;
+use Brain\Monkey\Functions;
+
+/**
+ * @covers \WP_Rocket\Subscriber\Cache\PurgeActionsSubscriber:purge_user_cache
+ * @group purge_actions
+ */
+class Test_PurgeUserCache extends TestCase {
+	use \Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;
+
+	/**
+	 * Test should not purge user cache when user cache is disabled
+	 */
+	public function testShouldNotPurgeUserCacheWhenUserCacheDisabled() {
+		add_filter( 'pre_get_rocket_option_cache_logged_user', '__return_false' );
+
+		Functions\expect( 'rocket_clean_user' )
+			->never();
```

Comment:
Instead of mocking the function, in an integrated tests, we want to validate that the user's cache is not purged when firing the `'delete_user'` event.

Why?

It validates the real expected behavior: when user cache is disabled (which happens by the filter), don't delete the user cache.

How?
Add a user cache. Then fire the event. Then check that the user's cache did not change.

Reply:
That was my original idea, but then here was my thoughts:
- I want to validate the behaviour of the `purge_user_cache()` specifically
- If I add a user cache, I'm doing more than that, I'm validating `rocket_clean_user()` as a side-effect

So I opted for mocking `rocket_clean_user()` there

Reply:
In a unit test, you are testing the isolated behavior of `purge_user_cache()`. Here you are testing to integrated behavior. Its job is to purge or not purge. The implementation of how we purge might change over time, meaning that `rocket_clean_user()` might be changed out to a different implementation. But the behavior and intent of the method stays the same: purge the user's cache.